lib/private/RedisFactory.php

```
@@ -78,7 +97,24 @@ private function create() {
 				$timeout = 0.0; // unlimited
 			}
 
-			@$this->instance->connect($host, $port, $timeout);
+			if (isset($config['connection_parameters'])) {
+				if (!$this->isConnectionParametersSupported()) {
+					throw new \Exception(\sprintf('Php-redis extension must be equal %s or higher to support connection parameters',
```

This should throw a `UnexpectedValueException` to match the check above. In fact, you could move this check to a private method to prevent code duplication.

---

\Exception changed to \UnexpectedValueException, not a fan of a function like isConnectionParametersSupportedAndThrowErrorIfNot